src/AutoRest.CSharp/Common/Output/Models/MethodSignature.cs

```diff
@@ -98,5 +105,31 @@ private MethodSignature MakeSync()
         }
 
         public FormattableString GetCRef() => $"{Name}({Parameters.GetTypesFormattable()})";
+
+        private class MethodSignatureParameterAndReturnTypeEqualityComparer : IEqualityComparer<MethodSignature>
+        {
+            public bool Equals(MethodSignature? x, MethodSignature? y)
+            {
+                if (ReferenceEquals(x, y))
+                {
+                    return true;
+                }
+
+                if (x is null || y is null)
+                {
+                    return false;
+                }
+
+                var result = x.Name == x.Name
+                    && x.ReturnType.EqualsByName(y.ReturnType)
```

Comment:
could this be `EqualsIgnoreNullable` instead of `EqualsByName`?
This `EqualsByName` is kind of a workaround for some special situation in MPG which should be removed - or promoted into the class of CSharpType if it is really needed somewhere

Reply:
Don't remember why I used `EqualsByName` in the first place, `Equals` works for now.

Reply:
There are other places that are using EqualsByName as well (other comparer defined in this PR), be sure to change them all

Reply:
There are other 2 places using `EqualsByName` can't be replaced by `Equals`, because the types having different implementations, which is compared in [Equals](https://github.com/Azure/autorest.csharp/blob/178d7ba7b164d702a2f66468e7e38ef8c0f10168/src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs#L118)

Reply:
Ah I see.
Could you move the EqualsByName to the `CSharpType` class? It is now an extension method.
Double confirm: this method also checks if they have the same namespace as well as the name?

Reply:
It doesn't, is namespace part of type name here?

Reply:
No, it's not, I will add namespace to the method.

Reply:
Updated